python/salpytools/states.py

```diff
@@ -167,102 +86,135 @@ def wake(self):
 
     #<----- Default State methods corresponding to UML design under here ------>
 
-    def disable(self):
+    def disable(self, model):
         raise StateTransitionException()
 
-    def enable(self):
+    def enable(self, model):
         raise StateTransitionException()
 
-    def exit(self):
+    def exit_control(self, model):
         raise StateTransitionException()
 
-    def go_to_standby(self):
+    def standby(self, model):
         raise StateTransitionException()
 
-    def start(self):
+    def start(self, model):
         raise StateTransitionException()
 
-    def enter_control(self):
+    def enter_control(self, model):
         raise StateTransitionException()
 
+
 class OfflineState(DefaultState):
 
-    def __init__(self, device, model, events=('SummaryState',), tsleep=0.5):
-        super(OfflineState, self).__init__('OFFLINE', device, model, events, tsleep)
+    def __init__(self, subsystem_tag, events=('SummaryState',), tsleep=0.5):
+        super(OfflineState, self).__init__('OFFLINE', subsystem_tag, events, tsleep)
+
+    def enter_control(self, model):
+        model.change_state("STANDBY")
+        self.send_logEvent("SummaryState", SummaryStateValue=5)
+
+    def exit(self, model):
+        print("Offline: exit() not implemented")
+
+    def do(self, model):
+        print("Offline: do() not implemented")
 
-    def enter_control(self):
-        self.model.change_state("STANDBY")
 
 class StandbyState(DefaultState):
 
-    def __init__(self, device, model, events=('SummaryState',), tsleep=0.5):
-        super(StandbyState, self).__init__('STANDBY', device, model, events, tsleep)
+    def __init__(self, subsystem_tag, events=('SummaryState',), tsleep=0.5):
+        super(StandbyState, self).__init__('STANDBY', subsystem_tag, events, tsleep)
 
-    def exit(self):
-        self.model.change_state("OFFLINE")
+    def exit_control(self, model):
+        model.change_state("OFFLINE")
+        self.send_logEvent("SummaryState", SummaryStateValue=4)
 
-    def start(self):
-        self.model.change_state("DISABLED")
+    def start(self, model):
+        model.change_state("DISABLED")
+        self.send_logEvent("SummaryState", SummaryStateValue=1)
+
+    def exit(self, model):
+        print("Standby: exit() not implemented")
```

Comment:
Please, use `self.log.debug` instead of `print`.

Reply:
done